xmtp_proto/src/api_client.rs

```diff
@@ -72,11 +73,9 @@ impl StdError for Error {
     }
 }
 
-pub trait XmtpApiSubscription {
-    fn is_closed(&self) -> bool;
-    fn get_messages(&self) -> Vec<Envelope>;
-    fn close_stream(&mut self);
-}
+pub trait XmtpApiSubscription {}
+
+impl<T: Stream<Item = Envelope>> XmtpApiSubscription for T {}
```

Comment:
Beyond the WASM implementation, this is conceptually the big change to `subscribe()` across both `grpc` and `grpc-gateway` clients. Now the result is just an ordinary `Stream` of `Envelope`s.

Reply:
Do we need the `XmtpApiSubscription` trait at all, seeing as it's empty? Can we just mandate that `subscribe()` returns a `Stream` instead of an `XmtpApiSubscription`? I think that lets us call the `Stream` methods on consumers even if the concrete type for `XmtpApiClient` is not known.